# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1481039060 18000
#      Tue Dec 06 10:44:20 2016 -0500
# Node ID 78f1d359dd8fbba5a954c5cb38ff0276f8d420f0
# Parent  b5cf18f8c18b53d9dc1fafc892c971fce7f3ba69
define octave_f77_int_type in octave-config.h (bug #49776)

* mk-octave-config-h.sh: Also copy octave_f77_int_type typedef to
octave-config.h.

diff --git a/build-aux/mk-octave-config-h.sh b/build-aux/mk-octave-config-h.sh
--- a/build-aux/mk-octave-config-h.sh
+++ b/build-aux/mk-octave-config-h.sh
@@ -148,19 +148,27 @@ EOF
 
 octave_idx_type="`$SED -n 's/#define OCTAVE_IDX_TYPE \([_a-zA-Z][_a-zA-Z0-9]*\)/\1/p' $config_h_file`"
 
 if test -z "$octave_idx_type"; then
   echo "mk-octave-config-h.sh: failed to find OCTAVE_IDX_TYPE in $config_h_file" 1>&2
   exit 1
 fi
 
+octave_f77_int_type="`$SED -n 's/#define OCTAVE_F77_INT_TYPE \([_a-zA-Z][_a-zA-Z0-9]*\)/\1/p' $config_h_file`"
+
+if test -z "$octave_f77_int_type"; then
+  echo "mk-octave-config-h.sh: failed to find OCTAVE_F77_INT_TYPE in $config_h_file" 1>&2
+  exit 1
+fi
+
 cat << EOF
 
 typedef $octave_idx_type octave_idx_type;
+typedef $octave_f77_int_type octave_f77_int_type;
 
 EOF
 
 if grep "#define OCTAVE_ENABLE_FLOAT_TRUNCATE 1" $config_h_file > /dev/null; then
   echo "#  define OCTAVE_FLOAT_TRUNCATE volatile"
 else
   echo "#  define OCTAVE_FLOAT_TRUNCATE"
 fi
